### ISSUE TITLE
Doesn't work if entry stylesheet is not in the main module

### ISSUE THREAD

**Issue by cprecioso** [cprecioso/parcel-plugin-purgecss#10]
The entry stylesheet must be in your package. That means that if you import a stylesheet of your npm dependencies directly from a JS or HTML file, it will not be purged. If you import it directly or indirectly from another stylesheet in your main package (or SASS, Stylus, etc), it's fine.

This is because `parentBundle.entryAsset.getConfig` will try to find `purgecss.config.js` in the entry CSS's module.

**Comment by cprecioso** [comment]
This could be alleviated if we checked if `getConfig` failed and then pass it the `path: process.cwd()` option.

**Comment by brianzelip** [comment]
FYI - this plugin **will** work if css is imported from node_modules via a Vue.js single-file-component `<style>` block, ie:

```vue
<template>
  <div>
    <h1>Hello world!</h1>
  </div>
</template>

<style>
@import "css-installed-via-package-json";
</style>



```

**Comment by cprecioso** [comment]
@brianzelip That is to be expected, as that part is handled by `CSSAsset`.

**Comment by SwiftWinds** [comment]
> That means that if you a stylesheet of your npm dependencies directly from a JS or HTML file, it will not be purged.

So if I import my Bootstrap and other CSS from node_modules in main.js like [this](https://github.com/Toga-Hacks-I/Toga-Hacks-Website/blob/master/js/main.js), will it not purge correctly? When running `npm run build`, I ran into a bunch of styles not being applied, but hopefully, I can solve that by finding all the classes that were toggled by my jQuery.

I just wanted to ask you this in case it was a different problem besides ignoring classes toggled by my JS

**Comment by cprecioso** [comment]
Yep, create a stylesheet in your source files and import all those stylesheets from there, instead of directly from the
JS file

**Comment by cprecioso** [comment]
Btw, this issue is not being solved right now since there’s a workaround and I’m waiting for Parcel v2 to be released.

**Comment by SwiftWinds** [comment]
> Yep, create a stylesheet in your source files and import all those stylesheets from there, instead of directly from the JS file

Hmm... I tried importing it to my [style.css](https://github.com/Toga-Hacks-I/Toga-Hacks-Website/blob/try-out-purgecss/css/style.css) file instead of my [main.js](https://github.com/Toga-Hacks-I/Toga-Hacks-Website/blob/try-out-purgecss/js/main.js) file, but nothing changed; the navbar styles are still not being applied. 

Maybe the only problem all along was that classes added by jQuery were being removed when they shouldn't be?

**Comment by cprecioso** [comment]
Sorry, I’m not understanding. Is the problem that your CSS is not being applied (maybe try w/o purgecss first), or that purgecss isn’t working, or that it’s removing everything?

**Comment by SwiftWinds** [comment]
It works without PurgeCSS, but when adding it, some CSS (i.e., the navbar) is not applied.

I actually got it working by ignoring a bunch of classes and id's that were added/removed by my jQuery ([here](https://github.com/Toga-Hacks-I/Toga-Hacks-Website/blob/try-out-purgecss/purgecss.config.js) is my purgecss.config.js) :)

Although it reduced the unused CSS from about 95% to 84%, it still seems like PurgeCSS is not purging a lot of CSS that is unnecessary. For example:

![Chrome coverage information](https://user-images.githubusercontent.com/12981958/67164256-49dfea80-f32d-11e9-8370-4072ed1870aa.png)

The bootstrap `.form_control` class is being preserved, yet I never even used a single form input anywhere on my page! o_0

Is there a solution to this, or is this to be expected?

**Comment by cprecioso** [comment]
Looks to me it's to be expected that those styles are kept, as you use them in the `index.html` file: https://github.com/Toga-Hacks-I/Toga-Hacks-Website/search?q=form-control&unscoped_q=form-control

**Comment by SwiftWinds** [comment]
Ahh yes, sorry I forgot that we used that 😅. Still, 84% is quite big; is there anything I can do to reduce that further?

Also, PurgeCSS is removing the scroll-lock and greyed-out background when I create a modal and some animations (namely when clicking on the F.A.Q. dropdowns and wow.js fade-ins as the user scrolls). Do you know why this might be happening? 

The deployment is https://togahacks.com in case you want to check out what is broken (and [here](https://github.com/Toga-Hacks-I/Toga-Hacks-Website) is the GitHub repo).

**Comment by cprecioso** [comment]
I recommend you ask those questions at the purgecss, I have only used it for a small weekend project and don’t know much about it. This project is a side-effect of that but I am not a user of it anymore.

**Comment by SwiftWinds** [comment]
Ahh, OK. Thanks for directing me in the right direction.
